File: src/day1.scala
import scala.io.Source
@main 
def part1() = 

    var list1: Vector[Int] = Vector();
    var list2: Vector[Int] = Vector();

    val filename = "/home/axel/Documents/games/aoc24/src/day1in.txt"
    val source = Source.fromFile(filename)
    for (line <- source.getLines()) do
       val lines = line.split(' ')
       list1 = list1.appended(lines(0).toInt);
       list2 = list2.appended(lines(3).toInt);
    
    source.close()

    list1 = list1.sorted()
    list2 = list2.sorted()
    var sum = 0
    // for(i <- list1) do 
    //     sum = sum + i*list2.count( _ == i)
    for(i <- list1.indices) do 
        if list1(i) > list2(i) then
            sum = sum + list1(i) -list2(i)
        else
            sum = sum + list2(i) -list1(i)
    println(sum)

File: src/aoc24/day1.rs
pub fn part1(lines: Vec<String>){
    let mut list1: Vec<i64> = vec![];
    let mut list2: Vec<i64> = vec![];
    // let mut sum: i64 = 0;

    // Print the lines
    for line in &lines {
        let v: Vec<i64> = line.split("   ").map(|x| x.parse().unwrap()).collect();
        list1.push(v[0]);
        list2.push(v[1]);
    }
    list1.sort_unstable();
    list2.sort_unstable();

    let sum: i64 = list1
        .iter()
        .zip(list2)
        .map(|(a,b)| (a-b).abs())
        .sum();


    println!("{}", sum);
}

pub fn part2(lines: Vec<String>){
    let mut list1: Vec<usize> = vec![];
    let mut list2: Vec<usize> = vec![];
    // let mut sum: usize = 0;

    // Print the lines
    for line in &lines {
        let v: Vec<usize> = line.split("   ").map(|x| x.parse().unwrap()).collect();
        list1.push(v[0]);
        list2.push(v[1]);
    }
    list1.sort_unstable();
    list2.sort_unstable();

    let sum: usize = list1
        .iter()
        .map(|x| x * list2
            .iter()
            .filter(|&&y| y == *x)
            .count())
        .sum();

    println!("{}", sum);
}